From 15e039ca1ca1af1f2c52a21d3571e9629c9580ef Mon Sep 17 00:00:00 2001
From: Lubomir Rintel <lkundrak@v3.sk>
Date: Mon, 4 Jun 2018 10:13:00 -0400
Subject: [PATCH] Try to run freerouting by its JPackage launcher script

It's a sin to touch the Java tool by its JAR file in Fedora.
---
 pcbnew/dialogs/dialog_freeroute_exchange.cpp | 23 +++++++++++++++++---
 1 file changed, 20 insertions(+), 3 deletions(-)

diff --git a/pcbnew/dialogs/dialog_freeroute_exchange.cpp b/pcbnew/dialogs/dialog_freeroute_exchange.cpp
index 3200de151..17e9e0543 100644
--- a/pcbnew/dialogs/dialog_freeroute_exchange.cpp
+++ b/pcbnew/dialogs/dialog_freeroute_exchange.cpp
@@ -82,7 +82,10 @@ void DIALOG_FREEROUTE::init()
     SetFocus();
     m_freeRouterFound = false;
 
-    wxFileName fileName( FindKicadFile( wxT( "freeroute.jar" ) ), wxPATH_UNIX );
+    wxFileName fileName( FindKicadFile( wxT( "freerouting" ) ), wxPATH_UNIX );
+
+    if( !fileName.FileExists() )
+        fileName.Assign ( FindKicadFile( wxT( "freeroute.jar" ) ), wxPATH_UNIX );
 
     if( fileName.FileExists() )
         m_freeRouterFound = true;
@@ -131,8 +134,22 @@ void DIALOG_FREEROUTE::OnLaunchButtonClick( wxCommandEvent& event )
             return;
     }
 
-    wxFileName jarfileName( FindKicadFile( wxT( "freeroute.jar" ) ), wxPATH_UNIX );
-    wxString command;
+    wxFileName jarfileName( FindKicadFile( wxT( "freerouting" ) ), wxPATH_UNIX );
+    wxString command = wxT( "" );
+
+    if( jarfileName.FileExists() )
+    {
+        command << wxChar( '"' ) << jarfileName.GetFullPath() << wxChar( '"' );
+        // add option to load the .dsn file
+        command << wxT( " -de " );
+        // add *.dsn full filename (quoted):
+        command << wxChar( '"' ) << dsnFile << wxChar( '"' );
+
+        ProcessExecute( command );
+        return;
+    } else {
+        jarfileName.Assign ( FindKicadFile( wxT( "freeroute.jar" ) ), wxPATH_UNIX );
+    }
 
     // Find the Java application on Windows.
     // Colud be no more needed since we now have to run only java, not java web start
-- 
2.17.1

